lacework/resource_lacework_integration_aws_eks_audit_log_test.go

```diff
@@ -0,0 +1,144 @@
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckIntegrationAwsEksAuditExists(resourceTypeAndName),
+					resource.TestCheckResourceAttr(resourceTypeAndName, "enabled", "true"),
+				),
+			},
+			{
+				Config: testAccIntegrationAwsEksAuditConfig(
+					false,
+				),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckIntegrationAwsEksAuditExists(resourceTypeAndName),
+					resource.TestCheckResourceAttr(resourceTypeAndName, "enabled", "false"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckIntegrationAwsEksAuditDestroy(s *terraform.State) error {
+	lacework := testAccProvider.Meta().(*api.Client)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != testAccIntegrationAwsEksAuditResourceType {
+			continue
+		}
+
+		response, err := lacework.Integrations.GetAws(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		for _, integration := range response.Data {
+			if integration.IntgGuid == rs.Primary.ID {
+				return fmt.Errorf("the AWS integration (%s) still exists", rs.Primary.ID)
+			}
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckIntegrationAwsEksAuditExists(resourceTypeAndName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		lacework := testAccProvider.Meta().(*api.Client)
+
+		rs, ok := s.RootModule().Resources[resourceTypeAndName]
+		if !ok {
+			return fmt.Errorf("resource (%s) not found", resourceTypeAndName)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("resource (%s) ID not set", resourceTypeAndName)
+		}
+
+		response, err := lacework.Integrations.GetAws(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		if len(response.Data) < 1 {
+			return fmt.Errorf("the AWS integration (%s) doesn't exist", rs.Primary.ID)
+		}
+
+		for _, integration := range response.Data {
+			if integration.IntgGuid == rs.Primary.ID {
+				return nil
+			}
+		}
+
+		return fmt.Errorf("the AWS integration (%s) doesn't exist", rs.Primary.ID)
+	}
+}
+
+func testAccIntegrationAwsEksAuditEnvVarsPreCheck(t *testing.T) {
+	if v := os.Getenv(testAccIntegrationAwsEnvSnsArn); v == "" {
+		t.Fatalf("%s must be set for acceptance tests", testAccIntegrationAwsEnvSnsArn)
+	}
+	if v := os.Getenv(testAccIntegrationAwsEnvRoleArn); v == "" {
+		t.Fatalf("%s must be set for acceptance tests", testAccIntegrationAwsEnvRoleArn)
+	}
+	if v := os.Getenv(testAccIntegrationAwsEnvExternalId); v == "" {
+		t.Fatalf("%s must be set for acceptance tests", testAccIntegrationAwsEnvExternalId)
+	}
+}
+
+func testAccIntegrationAwsEksAuditConfig(enabled bool) string {
+	return fmt.Sprintf(`
+resource "%s" "%s" {
+    name = "integration test"
+    enabled = %t
+    queue_url = %s
+    credentials {
+        role_arn = "%s"
+        external_id = "%s"
+    }
+}
+`,
+		testAccIntegrationAwsEksAuditResourceType,
+		testAccIntegrationAwsEksAuditResourceName,
+		enabled,
+		os.Getenv(testAccIntegrationAwsEnvSnsArn),
+		os.Getenv(testAccIntegrationAwsEnvRoleArn),
+		os.Getenv(testAccIntegrationAwsEnvExternalId),
```

Comment:
I know these values are being reused from ct test, but perhaps better create specifc eks audit log ones so the tests are a little more self contained